asterius/rts/rts.time.mjs

```diff
@@ -1,32 +1,52 @@
 import * as rtsConstants from "./rts.constants.mjs";
 
 export class TimeCBits {
-  constructor(memory) {
+  constructor(memory, targetSpecificModule) {
     this.memory = memory;
-    this.epoch = Date.now();
+    // Obtain Time API from the passed target-specific module
+    this.resolution = targetSpecificModule.Time.resolution; // ns
+    this.now = targetSpecificModule.Time.now; // output: ms
+    this.time = targetSpecificModule.Time.time; // output: [s,ns]
     Object.freeze(this);
   }
 
-  clock_getres(clk_id, res) {
-    if (res) {
-      this.memory.i64Store(res + rtsConstants.offset_timespec_tv_sec, 1);
-      this.memory.i64Store(res + rtsConstants.offset_timespec_tv_nsec, 0);
-    }
-    return 0;
+  /**
+   * Returns a (monotonic) nanoseconds timestamp.
+   */
+  getMonotonicNSec() {
+    // convert this.now() from ms to ns
+    return Math.floor(this.now() * 1000000);
   }
 
-  clock_gettime(clk_id, tp) {
-    if (tp) {
-      const ms = Date.now(),
-        s = Math.floor(ms / 1000),
-        ns = (ms - s * 1000) * 1000000;
-      this.memory.i64Store(tp + rtsConstants.offset_timespec_tv_sec, s);
-      this.memory.i64Store(tp + rtsConstants.offset_timespec_tv_nsec, ns);
+  /**
+   * Stores at a memory address the resolution of a given clock.
+   * @param clk_id the id of the clock
+   * @param addr the memory address 
+   */
+  clock_getres(clk_id, addr) {
```

Comment:
Now that we have `now` for monotonic clock and `time` for UTC clock, it would be nice to start using `clk_id` for both `clock_getres` and `clock_gettime` to know which clock we're querying, and use results from different underlying clocks. We need to know the constants for `CLOCK_MONOTONIC` and `CLOCK_REALTIME` here, and the rough steps to use those constants are:

1. Adding relevant constants in `cbits` of `ghc-toolkit`, and their foreign imports
2. Modify the `Asterius.JSGen.Constants` to emit those constants into `rts.constants.mjs` script, and run `genconstants` to generate new script to update the one in the project git tree
3. Import and use `rts.constants.mjs` file here